src/liballoc/tests/lib.rs

```diff
@@ -15,6 +15,7 @@
 #![feature(vec_remove_item)]
 #![feature(split_inclusive)]
 #![feature(binary_heap_retain)]
+#![allow(unknown_lints, confusable_idents)]
```

Comment:
Is there a reason we're applying these? Did I miss something in the PR description?

Reply:
There're actually some identifiers considered confusable in the code base. The most common case are `I` in generics to `l` as variable name, `rl` as `resolver_lifetime` to `r1` as variable name, etc.

To reduce noise in this PR i didn't change them within this PR, but deferred them, thus these `#![allow]`s. Will work on changing those and remove these `#![allow]`s once this PR lands.

Reply:
Oh. We will probably want to ship the lint as allow by default to start off with (we can make that switch at the very end so perf isn't affected). I suspect we'll not want it firing on `I` for example in generics, as it is very commonly used throughout rust codebases and in monospace (I guess technically serif?) fonts isn't very confusable.

Reply:
Good idea! Thanks!